tempesta_fw/t/functional/README.md

```diff
@@ -0,0 +1,197 @@
+to run testing framework on the same host. It is recommended to run testing
+framework on a separated host.
+
+Recommended test-beds:
+
+- Fully distributed. 4 different hosts with their own roles are used. This
+configuration isolates traffic generated by benchmark utilities and traffic
+generators in test network. It is also able to detect hangs and crushes. This
+configuration is the only recommended for performance and stress tests.
+```
+    ┌───────────────────┐
+    │ Testing Framework │
+    └──────┬────────────┘
+           │
+           │    Management over SSH
+           ├─────────────────┬─────────────────┐
+           │                 │                 │
+    ┌──────┴─────┐    ┌──────┴─────┐    ┌──────┴─────┐
+    │ Client     │    │ TempestaFW │    │ Web Server │
+    └──────┬─────┘    └──────┬─────┘    └──────┬─────┘
+           │                 │                 │
+           └─────────────────┴─────────────────┘
+            Separated network for test traffic
+```
+
+- With isolated testing framework. This preset have limited resources and
+results of the performance tests may vary between runs. But much less resource
+needed to run the tests.
+```
+    ┌───────────────────┐
+    │ Testing Framework │
+    └──────┬────────────┘
+           │
+           │ Management over SSH
+           │
+    ┌──────┴───────────────────────────┐
+    │ Client + TempestaFW + Web Server │
+    └──────────────────────────────────┘
+```
+
+It is also possible to run everything on the same host, but the configuration
+is unreliable.
+
+There is two different models of tests: workload tests and pure functional
+tests. Workload tests uses fully functional HTTP benchmark programs (ab, siege,
+wrk) and HTTP servers (Apache, nginx) to check TempestaFW behaviour. This type
+of tests is used for schedulers, stress and performance testing.
+
+Pure functional tests check internal logic. Here combined HTTP client-server
+server is used. It sends HTTP messages to TempestaFW, analyses how they are
+forwarded to server, and vice versa. [This tests are still under development.]
+
+
+## Requirements
+
+- Host for testing framework: `Pyhon3` and `pyhon3-paramiko`
+- All hosts except previous one:  `grep`, `awk`, `sftp-server`
+- Host for running clients: `wrk`
+- Host for running TempestaFW: Linux kernel with Tempesta, TempestaFW sources
+- Host for running server: `nginx`, web content directory accessible by nginx
+
+Testing framework manages other hosts via SSH protocol, so the host running
+testing framework must be able to be authenticated on other hosts by the key.
+That can be done using `ssh-copy-id`.
+
+
+## Run tests
+
+### Configuration
+
+Testing framework has a lot of options described in `tests_config.ini'.
+You can create default tests configuration by calling:
+
+```sh
+$ ./run_tests.py -d
+```
+Or run next command to show an example of configuration:
+```sh
+$ ./run_tests.py -e
+```
```

Comment:
The option is good, but a user still needs to read the README to configure the test suite. And that's not comfortable at all. It's better to add a default configuration file with comments for all the options.